actions/build-and-push-image-digests/action.yaml

```diff
@@ -0,0 +1,147 @@
+name: Build and push image digests to GitHub
+description: Composite action to build and push image digests to GitHub
+
+inputs:
+  registry:
+    description: |
+      Google Artifact Registry to store docker images in.
+    default: "us-docker.pkg.dev"
+  tag:
+    description: |
+      Docker images tag to be pushed.
+    required: true
+  context:
+    description: |
+      Path to the Docker build context.
+    default: "."
+  environment:
+    description: |
+      Environment for pushing artifacts (can be either dev or prod).
+    default: dev
+  image_name:
+    description: |
+      Name of the image to be pushed to GAR.
+    required: true
+  build-args:
+    description: |
+      List of arguments necessary for the Docker image to be built.
+    default: ""
+  file:
+    description: |
+      The dockerfile to use.
+    required: false
+  platform:
+    description: |
+      Platforms to build the image for
+    required: true
+  cache-from:
+    description: |
+      Where cache should be fetched from
+    required: false
+    default: "type=gha"
+  cache-to:
+    description: |
+      Where cache should be stored to
+    required: false
+    default: "type=gha,mode=max"
+  ssh:
+    description: |
+      List of SSH agent socket or keys to expose to the build
+  build-contexts:
+    description: |
+      List of additional build contexts (e.g., name=path)
+    required: false
+  docker-buildx-driver:
+    description: |
+      The driver to use for Docker Buildx
+    required: false
+    default: "docker-container"
+
+outputs:
+  full_image_name:
+    description: Full image name
+    value: ${{ steps.export-full-image-name.outputs.full_image_name }}
+
+runs:
+  using: composite
+  steps:
+    - name: try printing env
+      shell: bash
+      run: |
+        echo ${{ env.platform_env }}
+    - name: Prepare
+      shell: bash
+      run: |
+        platform=${{ inputs.platform }}
+        echo "PLATFORM_PAIR=${platform//\//-}" >> $GITHUB_ENV
+    - name: Checkout
+      uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
+      with:
+        repository: grafana/shared-workflows
+        ref: multiarch-docker-builds-using-sh-runners
+        path: shared-workflows
+    - name: Get repository name
+      id: get-repository-name
+      shell: bash
+      run: |
+        REPO_NAME=$(echo ${{ github.repository }} | awk -F'/' '{print $2}')
+        echo "repo_name=${REPO_NAME}" >> ${GITHUB_OUTPUT}
+    - name: Resolve GCP project
+      id: resolve-project
+      shell: bash
+      run: |
+        if [ "${{ inputs.environment }}" == 'dev' ]; then
+          PROJECT="grafanalabs-dev"
+        elif [ "${{ inputs.environment }}" == 'prod' ]; then
+          PROJECT="grafanalabs-global"
+        else
+          echo "Invalid environment. Valid environment variable inputs: dev, prod"
+          exit 1
+        fi
+        echo "project=${PROJECT}" >> ${GITHUB_OUTPUT}
```

Comment:
And this probably, or alternatively made into a standalone action since we repeat it all over the place.